client/components/import/import.js

```diff
@@ -135,6 +143,40 @@ ImportPopup.extendComponent({
   },
 
   getMethodName() {
+    return 'importCsvData';
+  },
+
+  getLabel() {
+    return 'import-tsv-instruction';
+  },
+
+  parseDataObject(text) {
+    if (window.Papa) {
+      // if the text contains '\t', we convert TSV to CSV
+      const csv = (text.indexOf('\t') > 0) ? text.replace(/(\t)/g, ',') : text;
+      const ret = window.Papa.parse(csv);
+      if (ret && ret.data && ret.data.length) return ret.data;
+      else throw new Meteor.Error('error-json-schema');
+    }
+  },
+
+  getDataPlaceholder() {
+    return 'import-tsv-placeholder';
+  },
+
+}).register('listImportCardsTsvPopup');
+
+ImportPopup.extendComponent({
+  getAdditionalData() {
+    const listId = this.currentData()._id;
+    const selector = `#js-list-${this.currentData()._id} .js-minicard:first`;
+    const firstCardDom = $(selector).get(0);
+    const sortIndex = Utils.calculateIndex(null, firstCardDom).base;
```

Comment:
Could you just add a `XXX comment` saying that we should find a way to avoid this crazy DOM reading gymnastic at some point? (This is currently the easiest way to retreive the index so keep the code untouched, I would just like a reminder when we refactor client side state stores).